draco/analysis/hyforesbandpass.py

```diff
@@ -0,0 +1,520 @@
+                    filt_f_f = np.ascontiguousarray(filt[pp,:,:,xx, tt])
+
+                    weight_mask_N = weight[pp, :, xx, tt] > 0.0 # this accounts for RFI, daytime, and other masks # does not account for aliasing
+                    vis_f_l_masked = vis_f_l*weight_mask_N[:, np.newaxis]*el_mask[np.newaxis, :] # apply the aliasing mask
+
+                    N[pp, xx, :, :] += np.matmul(np.conj(vis_f_l_masked), vis_f_l_masked.T)*filt_f_f
+                    # the line above does vis[pp,:, xx, :, tt].conj().dot(vis[pp,:, xx, :, tt].T())*filt[pp,:,:,xx, tt]
+
+                self.log.debug(f"Window summed for Polarization {pp} of {npol} and East-West baseline {xx} of {new}.")
+
+        self.log.debug(f"Window computation finished. Took {time.time() - t0:0.3f} seconds in total.")
+
+        # create variables to reduce the partial sums
+        yN_sum = np.zeros_like(yN)
+        N_sum = np.zeros_like(N)
+        D_sum = np.zeros_like(D)
+
+        self.comm.Allreduce(yN, yN_sum, op=MPI.SUM)
+        self.comm.Allreduce(N, N_sum, op=MPI.SUM)
+        self.comm.Allreduce(D, D_sum, op=MPI.SUM)
+
+        y = np.zeros_like(yN_sum)
+        y = yN_sum*tools.invert_no_zero(D_sum)
+        W = np.zeros_like(N_sum)
+        W = N_sum*tools.invert_no_zero(D_sum[:, :, :, np.newaxis])
+
+        # put the estimated gains and window in a container
+        bp_gain_win = containers.VisBandpassWindowBaseline(
+                pol= hv.pol,
+                ew = hv.ew,
+                freq= hv.freq,
+                )
+        bp_gain_win.bandpass[...] = y # new question: do I need [...] here? should check out other examples
+        bp_gain_win.window[...] = W
+
+        return bp_gain_win
+
+    def aliased_el_mask(self, hv):
+        """Return a mask for the el axis to mask out zenith angles beyond the aliased horizon
+        Computed using the maximum frequency in the hybrid beamformed visibilities.
+
+        Parameters
+        ----------
+        hv: HybridVisStream
+        Input container for which we want to provide a aliased region mask along the el axis
+
+        Returns
+        -------
+        mask: a mask to flag aliased regions along the el axis
+        """
+
+        freq = np.max(hv.freq)
+        horizon_limit = self.get_horizon_limit(freq)
+        el = hv.index_map["el"]
+        mask = np.abs(el) < horizon_limit
+
+        return mask
+
+
+    def get_horizon_limit(self, freq):
+        """Calculate the value of sin(za) where the southern horizon aliases.
+        Parameters
+        ----------
+        freq : np.ndarray[nfreq,]
+            Frequency in MHz.
+
+        Returns
+        -------
+        horizon_limit : np.ndarray[nfreq,]
+            This is the value of sin(za) where the southern horizon aliases.
+            Regions of sky where ``|sin(za)|`` is greater than or equal to
+            this value will contain aliases.
+        """
+        
+        return scipy.constants.c / (freq * 1e6 * self.min_ysep) - 1.0
```

Comment:
This is nitpicking, but for consistency we used `units.c` from `cora.util`, which is already imported in this module. Alternatively, it's ok if you use `scipy.constants.c`, but you should make sure that's used consistently in this module and then remove the `from cora.util import units` import